Automattic/jetpack
# Protect: Normalize status report data on the server side

This PR moves the normalization of the Protect status report data from the client side to the server. Currently, the Protect report data is fetched from wpcom and passed directly into the React app, where it is then mutated to fit the needs of the UI. With this change, all of that work is moved to the server side, and the React app receives the data it needs in the format it needs.

#### Changes proposed in this Pull Request:

* Migrates `mergeInstalledAndCheckedLists()` and `normalizeCoreInformation()` from the `useProtectData()` hook in the React app into `Status::merge_installed_and_checked_lists()` and `Status::normalize_core_information()` on the server side.
* Adds `Status::normalize_report_data()` which applies the migrated functions to the Protect status report data fetched from wpcom before passing it to the React app.

#### Jetpack product discussion



1201069996155224-as-1202242680206295/f

#### Does this pull request change what data or activity we track or use?



No.

#### Testing instructions:






* Check out this branch, boot up a Jetpack site, and open `/wp-admin/admin.php?page=jetpack-protect`
* Open Redux dev tools and validate the `status`.
* Validate the results from the Protect report are reflected properly in the UI. For example:
  * Install a new plugin and validate that the "This item was added to your site after the most recent scan" info icon is shown.
  * Validate that the main vulnerabilities message is correct according to the logic in `/projects/plugins/protect/src/js/components/vulnerabilities-list/empty.jsx:29`
  * Add a plugin with a vulnerability and ensure it is displayed correctly.
